diff --git a/cx_Freeze/command/bdist_rpm.py b/cx_Freeze/command/bdist_rpm.py
index 3aa48aba..2f919b41 100644
--- a/cx_Freeze/command/bdist_rpm.py
+++ b/cx_Freeze/command/bdist_rpm.py
@@ -18,6 +18,7 @@ from typing import ClassVar
 
 from setuptools import Command
 
+from cx_Freeze._compat import IS_CONDA
 from cx_Freeze.exception import (
     ExecError,
     FileError,
@@ -412,6 +413,10 @@ class bdist_rpm(Command):
             f"BuildArch: {platform.machine()}",
         ]
 
+        # Fix for conda
+        if IS_CONDA:
+            spec_file.append("%define debug_package %{nil}")
+
         # Workaround for #14443 which affects some RPM based systems such as
         # RHEL6 (and probably derivatives)
         vendor_hook = check_output(
diff --git a/tests/test_freezer.py b/tests/test_freezer.py
index c8b585cd..86c06181 100644
--- a/tests/test_freezer.py
+++ b/tests/test_freezer.py
@@ -13,6 +13,7 @@ from cx_Freeze import Executable, Freezer
 from cx_Freeze._compat import (
     BUILD_EXE_DIR,
     EXE_SUFFIX,
+    IS_CONDA,
     IS_MACOS,
     IS_MINGW,
     IS_WINDOWS,
@@ -103,7 +104,7 @@ def test_freezer_default_bin_includes(tmp_path: Path, monkeypatch) -> None:
 
     freezer = Freezer(executables=[Executable("hello.py")])
     if IS_MACOS:
-        if sys.version_info[:2] <= (3, 10):
+        if sys.version_info[:2] <= (3, 10) or IS_CONDA:
             expected = f"libpython{PYTHON_VERSION}.dylib"
         else:
             expected = f"Python.framework/Versions/{PYTHON_VERSION}/Python"
