diff --git a/setup.py b/setup.py
index cc07b48..228d2c2 100755
--- a/setup.py
+++ b/setup.py
@@ -65,8 +65,8 @@ class build_ext(distutils.command.build_ext.build_ext):
                 extra_args.extend(get_config_var("LOCALMODLIBS").split())
             # fix a bug using macOS on Github Actions
             if "--with-lto" in get_config_var("CONFIG_ARGS"):
-                extra_args.append("-flto")
-                extra_args.append("-Wl,-export_dynamic")
+                pass#extra_args.append("-flto")
+                #extra_args.append("-Wl,-export_dynamic")
             else:
                 extra_args.append("-s")
         self.compiler.link_executable(
